### PR TITLE
Ec ftir

### PR DESCRIPTION
**Pull request by ScottSoren**
This includes a reader and a rough start on techniques and plotters for EC-FTIR data analysis from Imperial College London.

The new code is demo'd here: https://github.com/ixdat/ixdat/blob/ec_ftir/development_scripts/reader_demonstrators/demo_opus_ftir_reader.py

### CODE REVIEW

**Comment by rietbj** [comment]
Create a plot that shows potential on the y axis and wavenumber on the x axis. This should allow you pick potential ranges and wavenumber ranges

**Comment by ScottSoren** [comment]
I've made the requested plotter! Hope it proves useful. Looking forward to the review.

File: development_scripts/reader_demonstrators/demo_opus_ftir_reader.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ecftir.plot_measurement()
+
+
+ecftir.plot_stacked_spectra(
+    dn=40,
+    xspan=[1050, 1400],
+    xspan_bg=[1050, 1060],
+    scale_factor=1.5,
+    # average=False,  # no averaging
+    average=5,  # average 5 spectra each side
```
[the remark this answers is not in the document]
Comment (ScottSoren) [reply development_scripts/reader_demonstrators/demo_opus_ftir_reader.py:57]: That's completely up to you! This demo script is just showing you which options are available.

File: src/ixdat/plotters/spectrum_plotter.py [hunk 2]
```diff
@@ -209,6 +212,163 @@ def plot_waterfall(
...
+        # If an xspan (for plotting) is given, cut all the vectors accordingly:
+        if xspan:
+            mask = np.logical_and(xspan[0] < x, x < xspan[-1])
+            x = x[mask]
+            for i, y_vec in enumerate(y_vec_list):
+                y_vec_list[i] = y_vec[mask]
+
+        # a list of y ranges is useful in all cases for figuring out how to scale the
+        # y-vectors onto the y axis, which a bit confusingly represents something else.
+        y_ranges = [max(y_vec) - min(y_vec) for y_vec in y_vec_list]
```
[the remark this answers is not in the document]
Comment (ScottSoren) [reply src/ixdat/plotters/spectrum_plotter.py:328]: Great catch! I've now replaced the list comprehension with the vectorized code as suggested.

**Comment by ScottSoren** [comment]
Thanks, @rietbj , for the review! I've answered both comments and implemented the vectorization you suggested. 
Merging :) 
This will be released in ixdat 0.2.9.

**Pull request merged**